tests/data/almanack/repo_setup/create_repo.py

```diff
@@ -156,51 +158,98 @@ def create_entropy_repositories(base_path: pathlib.Path) -> None:
 
 
 def repo_setup(
-    repo_path: pathlib.Path, files: dict, branch_name: str = "main"
+    repo_path: pathlib.Path,
+    files: list[dict],
+    branch_name: str = "main",
+    dates: Optional[list[datetime]] = None,
```

Comment:
Pretty neat that you can build up a commit history in a test git repo using just this function!

Since `files` is now a list of commits which contains a dict of files per commit, I wonder if it makes sense to change the name, e.g. to `commits`?

This one's debatable and may be more work than it's worth, but I see that `dates` is a parallel array that gets zipped with the `files` to produce dates for each commit. Perhaps it might make sense to have the commit dates be inlined in the `files` data structure, e.g. something like:
```
commits = [
  {'commit_date': Optional[datetime] = None, 'files': dict }, ...
]
```
where the current date is used for each if `commit_date` wasn't provided?

FWIW, I'm fine keeping this all as-is, since in your test cases it seems you're mostly adding files in a single commit and my suggestions above would make those test cases more verbose.

Reply:
Thanks good idea! I've added this as a subsequent commit and adjusted existing tests to utilize this data structure. Moving forward I imagine we might need or want to test this in other areas.